### DISCUSSION TITLE
Migratate to rules_python

### DISCUSSION THREAD

**Discussion by burkpojken** [bazel-contrib/rules_python#2376] [Q&A]
We have been using python from our development machines and would like to migrate to use the hermetic python rules. A problem is that we depend on other repositories that also use python from development machines.
How can we mix usage of python from the development machines and hermetic python rules?

I have tried this:

MODULE.bazel
```
bazel_dep(name = "rules_python", version = "0.37.2")
python = use_extension("@rules_python//python/extensions:python.bzl", "python")
python.toolchain(python_version = "3.12")
use_repo(python, "python_3_12")
use_repo(python, "python_versions")

pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
pip.parse(
    hub_name = "pip",
    python_version = "3.12",
    requirements_lock = "//bazeltools:bazel_pip_requirements_lock.txt",
)
use_repo(pip, "pip")
```

BUILD file in main repo
```
load("@python_3_12//:defs.bzl", "py_binary")
load("@pip//:requirements.bzl", "requirement")

py_binary(
  name = "bin",
    srcs = ["bin.py"],
    deps = [
        requirement("pyyaml"),
    ],
)
```
BUILD file in secondary repo
```
py_binary(
  name = "bin",
    srcs = ["bin.py"],
)
```

If I build with this the hermetic toolchain is always used, both in main and secondary repo.
If I add the following to MODULE.bazel
```
register_toolchains("@rules_python//python/runtime_env_toolchains:all")
```
Python from the development machines is always used.

I would like to have the hermetic toolchain to be used in the places where use 
```
load("@python_3_12//:defs.bzl", "py_binary")
```

**Comment by aignas** [comment]
Publishing wheels from other repositories into a private wheel registry is the best way for now to have a hybrid approach.
